### ISSUE TITLE
The numpy/doc is getting huge.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#24736]
Cloning `numpy/doc` for document generation currently downloads about 400 MB, which is not trivial. Doing a "blobless" clone makes no difference. Maybe we could experiment with `git lfs`, which is supported by github. See https://git-lfs.com/.

**Comment by seberg** [comment]
I think the problem with the generated html is that there are so many small files, no?  Maybe one could make those smaller, I am not sure.

**Comment by charris** [comment]
Updating the documentation doesn't change any of the previous documents, so none of those files need to be downloaded. All we do is add new documents and change a few links. That sounds about perfect for something like `git-lfs`. Well, unless we actually change some directory contents by making another document release. It would seem that the way to go about things would be to upload the new documentation and change a few links.

**Comment by rgommers** [comment]
This is kinda by design. I pointed this out as a downside of migration away from the doc server, and it was deemed an acceptable penalty. We can grow that `doc` repo to 1 GB, which is the limit. No one needs to clone it outside of a few core devs, so the pain seems limited enough to not bother messing with it now I think?

**Comment by melissawm** [comment]
From personal experience, I would advise against git-lfs. [In another repo we had a less than ideal experience with it](https://github.com/napari/napari/issues/4306) including incurring in extra costs inadvertently. It seems like there is another option in [git-annex](https://git-annex.branchable.com/) but I don't have any experience with it.

**Comment by rgommers** [comment]
I suggest doing nothing and closing this issue, because this is entirely expected and according to what we agreed when making the move. We can go 3-4 years from now without having to take any action here.

**Comment by rgommers** [comment]
Yes, it is. The complication is that it's not serving a single doc build, but the docs for every minor release. That's why it has to be committed, and the repo size grows with each release.
